src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -569,9 +569,16 @@ char* java_lang_String::as_quoted_ascii(oop java_string) {
   return result;
 }
 
+// Return a Symbol for the input Java string. A new Symbol is created if one
+// does not exist. If the length of the input string exceeds Symbol::max_length,
+// the string is truncated to Symbol::max_length.
```

Comment:
I don't think this is generally the correct way to approach this. We should only be using `String::as_symbol` when we know the String will fit into a Symbol - if it doesn't then we shouldn't be using this process. So any errors here should be fatal IMO. Each of the clients of this API should be examined to check valid usage i.e. it is okay for field/methods names due to 64K limit on those, but not for class/interface names - and certainly not for arbitrary strings like exception messages! Relatedly the usages of `java_lang_Throwable::detail_message` should also be examined.

Reply:
> I don't think this is generally the correct way to approach this. We should only be using String::as_symbol when we know the String will fit into a Symbol - if it doesn't then we shouldn't be using this process. So any errors here should be fatal IMO. Each of the clients of this API should be examined to check valid usage i.e. it is okay for field/methods names due to 64K limit on those, but not for class/interface names - and certainly not for arbitrary strings like exception messages! Relatedly the usages of java_lang_Throwable::detail_message should also be examined.

All the other direct usages of java_lang_String::as_symbol are okay, except one from java_lang_Throwable::detail_message. All other callers convert a type string into a Symbol. 

I changed java_lang_String::as_symbol to truncate the string for java_lang_Throwable::detail_message usage mainly. Instead we should `guarantee` the string length to be < Symbol::max_length in  java_lang_String::as_symbol.